src/utils/object-tree-node.ts

```diff
@@ -92,8 +92,8 @@ class ObjectTreeNode implements ProxyHandler {
     this.children = Object.create(null);
   }
 
-  toObject(): Record<string, any> {
-    return this.changes;
+  unwrap(): any {
+    return this.changes || this.content;
```

Comment:
I guess it doesn't support the case if there is a partial edit and you are accessing a higher-level object. Like in test `#set works for deep set and access` if you want to validate `styles.colors.main`, not `styles.colors.main.sync`.

Reply:
Perhaps we can pull from here to do so...

https://github.com/validated-changeset/validated-changeset/blob/43b65252a7d871f1d04e5a46b171f6798935b99f/src/index.ts#L904-L923